File: Application/src/main/java/com/shareconnect/ProfileAdapter.java
package com.shareconnect;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.google.android.material.button.MaterialButton;
import java.util.List;

public class ProfileAdapter extends RecyclerView.Adapter<ProfileAdapter.ProfileViewHolder> {
    private List<ServerProfile> profiles;
    private OnProfileClickListener listener;
    private ProfileManager profileManager;

    public interface OnProfileClickListener {
        void onProfileClick(ServerProfile profile);
        void onSetDefaultClick(ServerProfile profile);
        void onDeleteClick(ServerProfile profile);
    }

    public ProfileAdapter(List<ServerProfile> profiles, OnProfileClickListener listener, ProfileManager profileManager) {
        this.profiles = profiles;
        this.listener = listener;
        this.profileManager = profileManager;
    }

    @NonNull
    @Override
    public ProfileViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_profile, parent, false);
        return new ProfileViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProfileViewHolder holder, int position) {
        ServerProfile profile = profiles.get(position);
        holder.bind(profile);
    }

    @Override
    public int getItemCount() {
        return profiles.size();
    }

    class ProfileViewHolder extends RecyclerView.ViewHolder {
        private TextView textViewProfileName;
        private TextView textViewProfileUrl;
        private TextView textViewDefault;
        private ImageView imageViewServiceType;
        private TextView textViewServiceType;
        private MaterialButton buttonSetDefault;
        private MaterialButton buttonDelete;

        public ProfileViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewProfileName = itemView.findViewById(R.id.textViewProfileName);
            textViewProfileUrl = itemView.findViewById(R.id.textViewProfileUrl);
            textViewDefault = itemView.findViewById(R.id.textViewDefault);
            imageViewServiceType = itemView.findViewById(R.id.imageViewServiceType);
            textViewServiceType = itemView.findViewById(R.id.textViewServiceType);
            buttonSetDefault = itemView.findViewById(R.id.buttonSetDefault);
            buttonDelete = itemView.findViewById(R.id.buttonDelete);
        }

        public void bind(ServerProfile profile) {
            textViewProfileName.setText(profile.getName());
            textViewProfileUrl.setText(profile.getUrl() + ":" + profile.getPort());
            
            // Set service type icon and text
            setServiceTypeIcon(profile);
            textViewServiceType.setText(profile.getServiceTypeName(itemView.getContext()));
            
            // Check if this is the default profile
            ServerProfile defaultProfile = profileManager.getDefaultProfile();
            boolean isDefault = defaultProfile != null && defaultProfile.getId().equals(profile.getId());
            textViewDefault.setVisibility(isDefault ? View.VISIBLE : View.GONE);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onProfileClick(profile);
                    }
                }
            });

            buttonSetDefault.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onSetDefaultClick(profile);
                    }
                }
            });

            buttonDelete.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onDeleteClick(profile);
                    }
                }
            });
        }
        
        private void setServiceTypeIcon(ServerProfile profile) {
            switch (profile.getServiceType()) {
                case ServerProfile.TYPE_METUBE:
                    imageViewServiceType.setImageResource(R.drawable.ic_service_metube);
                    break;
                case ServerProfile.TYPE_TORRENT:
                    imageViewServiceType.setImageResource(R.drawable.ic_service_torrent);
                    break;
                case ServerProfile.TYPE_JDOWNLOADER:
                    imageViewServiceType.setImageResource(R.drawable.ic_service_jdownloader);
                    break;
                default:
                    imageViewServiceType.setImageResource(R.drawable.ic_service_metube);
                    break;
            }
        }
    }
}

File: Application/src/main/kotlin/com/shareconnect/ProfileAdapter.kt
package com.shareconnect

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.button.MaterialButton

class ProfileAdapter(
    private val profiles: List<ServerProfile>,
    private val listener: OnProfileClickListener,
    private val profileManager: ProfileManager
) : RecyclerView.Adapter<ProfileAdapter.ProfileViewHolder>() {

    interface OnProfileClickListener {
        fun onProfileClick(profile: ServerProfile)
        fun onSetDefaultClick(profile: ServerProfile)
        fun onDeleteClick(profile: ServerProfile)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProfileViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_profile, parent, false)
        return ProfileViewHolder(view)
    }

    override fun onBindViewHolder(holder: ProfileViewHolder, position: Int) {
        val profile = profiles[position]
        holder.bind(profile)
    }

    override fun getItemCount(): Int {
        return profiles.size
    }

    inner class ProfileViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val textViewProfileName: TextView = itemView.findViewById(R.id.textViewProfileName)
        private val textViewProfileUrl: TextView = itemView.findViewById(R.id.textViewProfileUrl)
        private val imageViewDefault: ImageView = itemView.findViewById(R.id.imageViewDefault)
        private val imageViewAuthenticated: ImageView = itemView.findViewById(R.id.imageViewAuthenticated)
        private val imageViewServiceType: ImageView = itemView.findViewById(R.id.imageViewServiceType)
        private val textViewServiceType: TextView = itemView.findViewById(R.id.textViewServiceType)
        private val buttonSetDefault: MaterialButton = itemView.findViewById(R.id.buttonSetDefault)
        private val buttonDelete: MaterialButton = itemView.findViewById(R.id.buttonDelete)

        fun bind(profile: ServerProfile) {
            textViewProfileName.text = profile.name
            textViewProfileUrl.text = profile.url + ":" + profile.port

            // Set service type icon and text
            setServiceTypeIcon(profile)
            textViewServiceType.text = profile.getServiceTypeName(itemView.context)

            // Check if this is the default profile
            val defaultProfile = profileManager.defaultProfile()
            val isDefault = defaultProfile != null && defaultProfile.id == profile.id
            imageViewDefault.visibility = if (isDefault) View.VISIBLE else View.GONE

            // Check if this profile has authentication
            val hasAuth = !profile.username.isNullOrEmpty() && !profile.password.isNullOrEmpty()
            imageViewAuthenticated.visibility = if (hasAuth) View.VISIBLE else View.GONE

            itemView.setOnClickListener {
                listener.onProfileClick(profile)
            }

            buttonSetDefault.setOnClickListener {
                listener.onSetDefaultClick(profile)
            }

            buttonDelete.setOnClickListener {
                listener.onDeleteClick(profile)
            }
        }

        private fun setServiceTypeIcon(profile: ServerProfile) {
            when (profile.serviceType) {
                ServerProfile.TYPE_METUBE -> imageViewServiceType.setImageResource(R.drawable.ic_service_metube)
                ServerProfile.TYPE_TORRENT -> imageViewServiceType.setImageResource(R.drawable.ic_service_torrent)
                ServerProfile.TYPE_JDOWNLOADER -> imageViewServiceType.setImageResource(R.drawable.ic_service_jdownloader)
                else -> imageViewServiceType.setImageResource(R.drawable.ic_service_metube)
            }
        }
    }
}